include/seastar/core/internal/pollable_fd.hh

```diff
@@ -68,13 +68,13 @@ public:
     /// blocked and it didn't return the short read/write. We anticipate that
     /// the next I/O will also be non-blocking and will not return EAGAIN.
     /// But the speculation is invalidated once it is "used" by
-    /// \c take_speculation()
-    void speculate_epoll(int events) { events_known |= events; }
+    /// \c revoke_speculation()
+    void take_speculation(int events) { events_known |= events; }
```

Comment:
take_speculation() is a dangerous rename, since it already exists. How did you verify you didn't forget a call site?

Reply:
I suggest to rename it to `speculate()`, it won't confuse developers familiar with older versions of the code base.

Reply:
> take_speculation() is a dangerous rename, since it already exists.

Yes, this is also my concern. 

What I thought initialily is that since the two operations are opposite, I want their names to reflect it(using the `OP_speculation` style).

And I have considered other choices(like `do_speculation`, but it sounds like an internal method) but I don't find a more satisfactory one.

> How did you verify you didn't forget a call site?

I renamed the original `take_speculation` first and then `speculate_epoll`, and used the rename utility provided by IDE to ensure all call sites are updated.